pacman/model/partitioner_interfaces/abstract_splitter_common.py

```diff
@@ -0,0 +1,180 @@
+        for vertex in self._governed_app_vertex.machine_vertices:
+            result[vertex] = edge_types
+        return result
+
+    def set_max_atoms_per_core(self, max_atoms_per_core, is_fixed_atoms):
+        """ sets max atoms per core for this splitter object
+
+        :param max_atoms_per_core: max atoms per core for this splitter.
+        :param is_fixed_atoms: is this a hard constraint or soft.
+        :rtype: None
+        """
+
+        self._max_atoms_per_core = max_atoms_per_core
+        self._is_fixed_atoms_per_core = is_fixed_atoms
+
+    def set_governed_app_vertex(self, app_vertex):
+        """ sets a app vertex to be governed by this splitter object.
+        Once set it cant be reset
+
+        :param app_vertex: the app vertex to govern
+        :rtype: None
+        :raises PacmanConfigurationException if the app vertex has already\
+            been set.
+        """
+        if self._governed_app_vertex == app_vertex:
+            return
+        if self._governed_app_vertex is not None:
+            raise PacmanConfigurationException(
+                self.SETTING_SPLITTER_OBJECT_ERROR_MSG.format(
+                    self._governed_app_vertex, self._splitter_name,
+                    app_vertex))
+        self._governed_app_vertex = app_vertex
+        self.check_supported_constraints()
+        app_vertex.splitter_object = self
+
+    def check_supported_constraints(self):
+        utility_calls.check_algorithm_can_support_constraints(
+            constrained_vertices=[self._governed_app_vertex],
+            supported_constraints=[
+                MaxVertexAtomsConstraint, FixedVertexAtomsConstraint],
+            abstract_constraint_type=AbstractPartitionerConstraint)
+
+    def split(self, resource_tracker, machine_graph):
+        """ executes splitting
+
+        :param ResourceTracker resource_tracker: machine resources
+        :param MachineGraph machine_graph: machine graph
+        :return: bool true if successful, false otherwise
+        """
+        return self.create_machine_vertices(resource_tracker, machine_graph)
+
+    @abstractmethod
+    def create_machine_vertices(self, resource_tracker, machine_graph):
+        """ method for specific splitter objects to use.
+
+        :param resource_tracker:
+        :param machine_graph:
+        :return: bool true if successful, false otherwise
+        """
+
+    @abstractmethod
+    def get_out_going_slices(self):
+        """ allows a application vertex to control the set of slices for \
+        outgoing application edges
+        :return: list of Slices and bool of estimate or not
+        """
+
+    @abstractmethod
+    def get_in_coming_slices(self):
+        """ allows a application vertex to control the set of slices for \
+        incoming application edges
+        :return: the slices incoming to this vertex, bool if estimate or exact
+        """
+
+    @abstractmethod
+    def get_pre_vertices(self, edge, outgoing_edge_partition):
```

Comment:
I suggest changing method name to getget_outgoing_vertices.
1. Matches get_out_going_slices naming
2. The name keeps confuing even @alan-stokes

Reply:
I find it hard to see how some Vertices returned my this method could support some edge types while other returned by the SAME call could support other edge types.

Consider changing the return to a tuple of (edgeTypes and Vertices).

Reply:
I object to `getget`… 😉